core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -46,12 +48,18 @@
 
   private static final String METADATA_FOLDER_NAME = "metadata";
 
+  private String fullName;
   private TableMetadata currentMetadata = null;
   private String currentMetadataLocation = null;
   private boolean shouldRefresh = true;
   private int version = -1;
 
-  protected BaseMetastoreTableOperations() {
+  protected BaseMetastoreTableOperations(String fullName) {
```

Comment:
This base class is extended by external projects, so I think we should avoid breaking changes like this one.

I propose adding a `tableName` method with a base implementation that returns `null`. Hive can override the implementation to provide the table name (as could external implementations) and the log message can include the table name if it is non-null. Does that sound reasonable?

Reply:
Absolutely!
Done